bundle-workflow/src/test_workflow/dependency_installer.py

```diff
@@ -0,0 +1,47 @@
+import os
+import shutil
+
+
+class DependencyInstaller:
+    """
+    Provides functionality to copy the maven dependencies from S3 to maven local.
+    """
+
+    def __init__(self, build_id, dependency_name, version, arch):
+        self.build_id = build_id
+        self.dependency_name = dependency_name
+        self.version = version
+        self.arch = arch
+        self.dependency_path = f"org/opensearch/{self.dependency_name}/{self.version}/"
+        self.maven_local_path = os.path.join(
```

Comment:
i'm not sure what you mean by workers reuse. What I'm saying here is the workers would sync the dependencies in ~/.m2/repository of jenkins host. If there are two concurrent test jobs running, one for 1.1 another for 1.0, since they filesystem is same, they might end up overwriting each others dependencies.

Reply:
Added build-id in the path to keep the executions independent.